src/builder_state.rs

```diff
@@ -528,31 +565,35 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
             // add the local builder commitment list
             self.builder_commitments.push(builder_hash.clone());
 
-            //let num_txns = <TYPES::BlockPayload as TestBlockPayload>::txn_count(&payload);
             let encoded_txns: Vec<u8> = payload.encode().unwrap().collect();
-
             let block_size: u64 = encoded_txns.len() as u64;
-
             let offered_fee: u64 = 0;
 
             // get the total nodes from the builder state.
             // stored while processing the DA Proposal
             let vid_num_nodes = self.total_nodes.get();
 
-            // convert vid_num_nodes to usize
-            // spawn a task to calculate the VID commitment, and pass the builder handle to the global state
+            // spawn a task to calculate the VID commitment, and pass the handle to the global state
             // later global state can await on it before replying to the proposer
-            let join_handle =
-                task::spawn(async move { vid_commitment(&encoded_txns, vid_num_nodes) });
+            let (unbounded_sender, unbounded_reciever) = unbounded();
+            // #[allow(clippy::let_and_return)]
+            // let handle = async_spawn(async move {
+            //     let vidc = vid_commitment(&encoded_txns, vid_num_nodes);
+            //     vidc
+            // });
+            #[allow(unused_must_use)]
+            async_spawn(async move {
+                let vidc = vid_commitment(&encoded_txns, vid_num_nodes);
+                unbounded_sender.send(vidc).await;
+            });
 
-            //let signature = self.builder_keys.0.sign(&block_hash);
             return Some(BuildBlockInfo {
```

Comment:
Just 
```rust
            Some(BuildBlockInfo {
                builder_hash,
                block_size,
                offered_fee,
                block_payload: payload,
               metadata,
                vid_receiver: unbounded_reciever,
            })
```, without the return and semicolon

Reply:
https://github.com/EspressoSystems/hs-builder-core/pull/68/commits/832462931d7160e8b11e661e9859098fd5f20cc3